cloud/services/domains.go

```diff
@@ -62,6 +73,74 @@ func EnsureDNSEntries(ctx context.Context, mscope *scope.MachineScope, operation
 	return nil
 }
 
+// EnsureAkamaiDNSEntries ensures the domainrecord on Akamai EDGE DNS is created, updated, or deleted based on operation passed
+func EnsureAkamaiDNSEntries(ctx context.Context, mscope *scope.MachineScope, operation string, dnsEntries []DNSOptions) error {
+	// SetUp Client for Akamai EDGE DNS
+
+	fqdn := mscope.LinodeCluster.ObjectMeta.Name + "-" + mscope.LinodeCluster.Spec.Network.DNSUniqueIdentifier + "." + mscope.LinodeCluster.Spec.Network.DNSRootDomain
+
+	for _, dnsEntry := range dnsEntries {
+		recordBody, err := mscope.AkamaiDomainsClient.GetRecord(ctx, mscope.LinodeCluster.Spec.Network.DNSRootDomain, fqdn, string(dnsEntry.DNSRecordType))
+		if err != nil {
+			if operation == "create" {
+				if err := createAkamaiDNSEntry(ctx, mscope, fqdn, dnsEntry); err != nil {
+					return err
+				}
+			}
+			continue
+		}
+		if operation == "delete" {
+			switch {
+			case len(recordBody.Target) > 1:
+				recordBody.Target = removeElement(recordBody.Target, strings.Replace(dnsEntry.Target, "::", ":0:0:", 8)) //nolint:mnd // 8 for 8 octest
+				if err := updateAkamaiDNSEntry(ctx, mscope, recordBody); err != nil {
+					return err
+				}
+				continue
+			default:
+				if err := deleteAkamaiDNSEntry(ctx, mscope, recordBody); err != nil {
+					return err
+				}
+			}
+		} else {
+			recordBody.Target = append(recordBody.Target, dnsEntry.Target)
+			if err := updateAkamaiDNSEntry(ctx, mscope, recordBody); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func removeElement(stringList []string, elemToRemove string) []string {
```

Comment:
I think we might be able to instead use [slices.Delete](https://pkg.go.dev/golang.org/x/exp/slices#Delete) like how we already use [slices.Insert in the linodemachine_controller_helpers](https://github.com/linode/cluster-api-provider-linode/blob/main/controller/linodemachine_controller_helpers.go#L109)